### PR TITLE
Feature: Ability to change marking spam emails as "Read"

### PR DESCRIPTION
**Pull request by Riscue**
## What type of PR?

Feature

## What does this PR do?

Changes the default behavior of marking all incoming spam as read, giving the user a chance to decide. Parameter set to True default. Nothing will be changed unless user uncheck it.

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
Thank you for your contribution. Can you add a towncrier/newsfragment to the PR please (for the changelog)?

**Comment by Riscue** [comment]
> Thank you for your contribution. Can you add a towncrier/newsfragment to the PR please (for the changelog)?

Added changelog to PR. Thanks.

**Comment by ghostwheel42** [comment]
Hmm. Alembic didn't update my database when testing - I'll try again in the next days.
We'll also need some translations for this.

**Review by nextgens — DISMISSED** [review]
LGTM

File: core/admin/migrations/versions/8f9ea78776f4_.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# revision identifiers, used by Alembic.
+revision = '8f9ea78776f4'
+down_revision = '3b7eee912b41'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('user', sa.Column('spam_mark_as_read', sa.Boolean(), nullable=False))
```
Comment (ghostwheel42) [inline core/admin/migrations/versions/8f9ea78776f4_.py:18]: This yields an error:
```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) Cannot add a NOT NULL column with default value NULL
[SQL: ALTER TABLE user ADD COLUMN spam_mark_as_read BOOLEAN NOT NULL]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```
Please add the default: `server_default=sa.sql.expression.false()`

**Review by ghostwheel42 — DISMISSED** [review]
Looks good, but upgrading fails.

**Review by ghostwheel42 — APPROVED** [review]
lgtm

**Review by nextgens — APPROVED** [review]

**Comment by Riscue** [comment]
@ghostwheel42 @nextgens will this pr be included in any release soon?

**Comment by nextgens** [comment]
It's merged into master... so the next release will have it.

**Comment by Riscue** [comment]
Merged into master 6 months ago and several release passed since but none of them has it. Thanks for reply. I'll wait.

**Pull request merged**